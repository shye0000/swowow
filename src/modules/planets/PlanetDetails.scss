.planet-details {
  color: white;
  height: 100%;
  overflow-y: auto;
  overflow-x: hidden;
  padding: 20px 30px;
  .error {
    font-size: 6em;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
  .details-body {
    .sphere-wrapper {
      position: absolute;
      top: 10%;
      left: 50%;
      transform: translateX(-50%);
      z-index: -1;
    }
    .planet-name {
      margin-top: 30px;
      font-size: 2em;
    }
    .info-card {
      border: 1px solid white;
      margin-top: 30px;
      background: rgba(0,0,0,0.5);
      color: white;
      font-weight: bold;
      .ant-card-head {
        background: none;
        border-bottom: 1px solid white;
        .ant-card-head-title {
          color: white;
          font-weight: bold;
        }
      }
    }
  }
}