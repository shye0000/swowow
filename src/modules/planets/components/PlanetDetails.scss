@import "../../../styles/vars";
.planet-details {
  color: white;
  height: 100%;
  overflow-y: auto;
  overflow-x: hidden;
  padding: $sm-padding $lg-padding;
  .error {
    font-size: 6em;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
  .details-body {
    .sphere-wrapper {
      position: absolute;
      bottom: 15%;
      left: 50%;
      transform: translateX(-50%);
      z-index: -1;
    }
    .planet-name {
      margin-top: $lg-padding;
      font-size: 2em;
    }
    .info-card {
      border: 1px solid white;
      margin-top: $lg-padding;
      background: rgba(0,0,0,0.5);
      color: white;
      font-weight: bold;
      .ant-card-head {
        background: none;
        border-bottom: 1px solid white;
        .ant-card-head-title {
          color: white;
          font-weight: bold;
        }
      }
    }
  }
}