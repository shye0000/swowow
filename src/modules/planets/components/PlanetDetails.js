import React, {useEffect} from 'react';
import './PlanetDetails.scss';
import {slowDown, speedUp} from '../../../redux/actions';
import {fetchPlanet} from '../redux/actions';
import HomeLink from '../../../components/HomeLink';
import PlanetsTableLink from './PlanetsTableLink';
import Row from 'antd/lib/row';
import Col from 'antd/lib/col';
import Card from 'antd/lib/card';
import Sphere from './Sphere';
import PlanetBasicInfo from './PlanetBasicInfo';
import {connect} from 'react-redux';
import Animate from 'rc-animate';

const PlanetDetails = props => {
	const {
		currentPlanet,
		fetching,
		fetchSuccess,
		match,
		fetchPlanet,
	} = props;

	useEffect(() => {
		const {planetUrl} = match.params;
		const url = decodeURIComponent(planetUrl);
		let currentPlanetUrl;
		if (currentPlanet) {
			currentPlanetUrl = currentPlanet.url;
		}
		if (currentPlanetUrl !== url) {
			fetchPlanet(url);
		}
	});

	let content;
	if (fetching) {
		content = null;
	} else if (!fetchSuccess){
		content = <div className="error" >Bad luck, something went wrong...</div>;
	} else if (currentPlanet){
		const {diameter, name} = currentPlanet;
		content = <Animate
			transitionName="fade"
			transitionAppear
		>
			<div className="details-body">
				<div className="sphere-wrapper">
					<Sphere className="planet-sphere" diameter={diameter} />
				</div>
				<div className="planet-name">{name}</div>
				<Row gutter={24} type="flex">
					<Col xs={24} md={8}>
						<Card className="info-card" title="Information">
							<PlanetBasicInfo planet={currentPlanet}/>
						</Card>
					</Col>
					<Col xs={24} md={8}>
						<Card loading={true} className="info-card" title="Residents" />
					</Col>
					<Col xs={24} md={8}>
						<Card loading={true} className="info-card" title="Films" />
					</Col>
				</Row>
			</div>
		</Animate>;
	}
	return <div className="planet-details">
		<div><HomeLink /></div>
		<div><PlanetsTableLink /></div>
		{content}
	</div>;
};

const mapStateToProps = (state) => {
	return {
		fetching: state.planets.fetching,
		fetchSuccess: state.planets.fetchSuccess,
		currentPlanet: state.planets.currentPlanet
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		fetchPlanet: (planetUrl) => {
			dispatch(speedUp());
			dispatch(fetchPlanet(planetUrl)).then(() => dispatch(slowDown())).catch(() => dispatch(slowDown()));
		}
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(PlanetDetails);