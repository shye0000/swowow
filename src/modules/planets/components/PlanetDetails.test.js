import React from 'react';
import { PlanetDetails } from './PlanetDetails';
import { shallow } from 'enzyme';

describe('PlanetDetails', () => {

	it('render correctly', () => {
		const props = {
			fetching: false,
			fetchSuccess: true,
			currentPlanet: {url: 'planet_url'},
			match: {
				params: {planetUrl: 'planet_url'}
			}
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('when props.fetching "true", no error and no details-body rendered', () => {
		const props = {
			fetching: true,
			fetchSuccess: true,
			currentPlanet: {url: 'planet_url'},
			match: {
				params: {planetUrl: 'planet_url'}
			}
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		expect(wrapper.find('.error')).toHaveLength(0);
		expect(wrapper.find('.details-body')).toHaveLength(0);
	});

	it('when props.currentPlanet has value and match with the url encoded in route params, details-body rendered', () => {
		const props = {
			fetching: false,
			fetchSuccess: true,
			currentPlanet: {url: 'planet_url'},
			match: {
				params: {planetUrl: 'planet_url'}
			}
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		expect(wrapper.find('.details-body')).toHaveLength(1);
	});

	it('when props.fetchSuccess "false" and props.fetching "false", render error', () => {
		const props = {
			fetching: false,
			fetchSuccess: false,
			currentPlanet: {url: 'planet_url'},
			match: {
				params: {planetUrl: 'planet_url'}
			}
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		expect(wrapper.find('.error')).toHaveLength(1);
	});

	it('when props.currentPlanet has value but not match with route param, call fetchPlanet', () => {
		const props = {
			fetching: false,
			fetchSuccess: true,
			currentPlanet: {url: 'planet_url_not_same'},
			match: {
				params: {planetUrl: 'planet_url'}
			},
			fetchPlanet: jest.fn(() => {})
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		const componentDidMountSpy = jest.spyOn(wrapper.instance(), 'componentDidMount');
		componentDidMountSpy();
		expect(props.fetchPlanet).toBeCalled();
	});

	it('when props.currentPlanet no value, call fetchPlanet', () => {
		const props = {
			fetching: false,
			fetchSuccess: true,
			currentPlanet: null,
			match: {
				params: {planetUrl: 'planet_url'}
			},
			fetchPlanet: jest.fn(() => {})
		};
		const wrapper = shallow(<PlanetDetails {...props} />);
		const componentDidMountSpy = jest.spyOn(wrapper.instance(), 'componentDidMount');
		componentDidMountSpy();
		expect(props.fetchPlanet).toBeCalled();
	});

});