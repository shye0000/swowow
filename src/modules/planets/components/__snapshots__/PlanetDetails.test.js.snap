// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetDetails render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetDetails
    currentPlanet={
      Object {
        "url": "planet_url",
      }
    }
    fetchSuccess={true}
    fetching={false}
    match={
      Object {
        "params": Object {
          "planetUrl": "planet_url",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <HomeLink />
        </div>,
        <div>
          <PlanetsTableLink />
        </div>,
        <Animate
          animation={Object {}}
          component="span"
          componentProps={Object {}}
          onAppear={[Function]}
          onEnd={[Function]}
          onEnter={[Function]}
          onLeave={[Function]}
          transitionAppear={true}
          transitionEnter={true}
          transitionLeave={true}
          transitionName="fade"
        >
          <div
            className="details-body"
          >
            <div
              className="sphere-wrapper"
            >
              <Sphere
                className="planet-sphere"
                diameter={undefined}
              />
            </div>
            <div
              className="planet-name"
            />
            <Row
              gutter={24}
              type="flex"
            >
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  title="Information"
                >
                  <PlanetBasicInfo
                    planet={
                      Object {
                        "url": "planet_url",
                      }
                    }
                  />
                </Card>
              </Col>
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  loading={true}
                  title="Residents"
                />
              </Col>
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  loading={true}
                  title="Films"
                />
              </Col>
            </Row>
          </div>
        </Animate>,
      ],
      "className": "planet-details",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <HomeLink />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PlanetsTableLink />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animation": Object {},
          "children": <div
            className="details-body"
          >
            <div
              className="sphere-wrapper"
            >
              <Sphere
                className="planet-sphere"
                diameter={undefined}
              />
            </div>
            <div
              className="planet-name"
            />
            <Row
              gutter={24}
              type="flex"
            >
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  title="Information"
                >
                  <PlanetBasicInfo
                    planet={
                      Object {
                        "url": "planet_url",
                      }
                    }
                  />
                </Card>
              </Col>
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  loading={true}
                  title="Residents"
                />
              </Col>
              <Col
                md={8}
                xs={24}
              >
                <Card
                  className="info-card"
                  loading={true}
                  title="Films"
                />
              </Col>
            </Row>
          </div>,
          "component": "span",
          "componentProps": Object {},
          "onAppear": [Function],
          "onEnd": [Function],
          "onEnter": [Function],
          "onLeave": [Function],
          "transitionAppear": true,
          "transitionEnter": true,
          "transitionLeave": true,
          "transitionName": "fade",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="sphere-wrapper"
              >
                <Sphere
                  className="planet-sphere"
                  diameter={undefined}
                />
              </div>,
              <div
                className="planet-name"
              />,
              <Row
                gutter={24}
                type="flex"
              >
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    title="Information"
                  >
                    <PlanetBasicInfo
                      planet={
                        Object {
                          "url": "planet_url",
                        }
                      }
                    />
                  </Card>
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Residents"
                  />
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Films"
                  />
                </Col>
              </Row>,
            ],
            "className": "details-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Sphere
                  className="planet-sphere"
                  diameter={undefined}
                />,
                "className": "sphere-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "planet-sphere",
                  "diameter": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "planet-name",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      title="Information"
                    >
                      <PlanetBasicInfo
                        planet={
                          Object {
                            "url": "planet_url",
                          }
                        }
                      />
                    </Card>
                  </Col>,
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      loading={true}
                      title="Residents"
                    />
                  </Col>,
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      loading={true}
                      title="Films"
                    />
                  </Col>,
                ],
                "gutter": 24,
                "type": "flex",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Card
                      className="info-card"
                      title="Information"
                    >
                      <PlanetBasicInfo
                        planet={
                          Object {
                            "url": "planet_url",
                          }
                        }
                      />
                    </Card>,
                    "md": 8,
                    "xs": 24,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <PlanetBasicInfo
                        planet={
                          Object {
                            "url": "planet_url",
                          }
                        }
                      />,
                      "className": "info-card",
                      "title": "Information",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "planet": Object {
                          "url": "planet_url",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Card
                      className="info-card"
                      loading={true}
                      title="Residents"
                    />,
                    "md": 8,
                    "xs": 24,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "info-card",
                      "loading": true,
                      "title": "Residents",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Card
                      className="info-card"
                      loading={true}
                      title="Films"
                    />,
                    "md": 8,
                    "xs": 24,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "info-card",
                      "loading": true,
                      "title": "Films",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <HomeLink />
          </div>,
          <div>
            <PlanetsTableLink />
          </div>,
          <Animate
            animation={Object {}}
            component="span"
            componentProps={Object {}}
            onAppear={[Function]}
            onEnd={[Function]}
            onEnter={[Function]}
            onLeave={[Function]}
            transitionAppear={true}
            transitionEnter={true}
            transitionLeave={true}
            transitionName="fade"
          >
            <div
              className="details-body"
            >
              <div
                className="sphere-wrapper"
              >
                <Sphere
                  className="planet-sphere"
                  diameter={undefined}
                />
              </div>
              <div
                className="planet-name"
              />
              <Row
                gutter={24}
                type="flex"
              >
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    title="Information"
                  >
                    <PlanetBasicInfo
                      planet={
                        Object {
                          "url": "planet_url",
                        }
                      }
                    />
                  </Card>
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Residents"
                  />
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Films"
                  />
                </Col>
              </Row>
            </div>
          </Animate>,
        ],
        "className": "planet-details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <HomeLink />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PlanetsTableLink />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animation": Object {},
            "children": <div
              className="details-body"
            >
              <div
                className="sphere-wrapper"
              >
                <Sphere
                  className="planet-sphere"
                  diameter={undefined}
                />
              </div>
              <div
                className="planet-name"
              />
              <Row
                gutter={24}
                type="flex"
              >
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    title="Information"
                  >
                    <PlanetBasicInfo
                      planet={
                        Object {
                          "url": "planet_url",
                        }
                      }
                    />
                  </Card>
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Residents"
                  />
                </Col>
                <Col
                  md={8}
                  xs={24}
                >
                  <Card
                    className="info-card"
                    loading={true}
                    title="Films"
                  />
                </Col>
              </Row>
            </div>,
            "component": "span",
            "componentProps": Object {},
            "onAppear": [Function],
            "onEnd": [Function],
            "onEnter": [Function],
            "onLeave": [Function],
            "transitionAppear": true,
            "transitionEnter": true,
            "transitionLeave": true,
            "transitionName": "fade",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="sphere-wrapper"
                >
                  <Sphere
                    className="planet-sphere"
                    diameter={undefined}
                  />
                </div>,
                <div
                  className="planet-name"
                />,
                <Row
                  gutter={24}
                  type="flex"
                >
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      title="Information"
                    >
                      <PlanetBasicInfo
                        planet={
                          Object {
                            "url": "planet_url",
                          }
                        }
                      />
                    </Card>
                  </Col>
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      loading={true}
                      title="Residents"
                    />
                  </Col>
                  <Col
                    md={8}
                    xs={24}
                  >
                    <Card
                      className="info-card"
                      loading={true}
                      title="Films"
                    />
                  </Col>
                </Row>,
              ],
              "className": "details-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Sphere
                    className="planet-sphere"
                    diameter={undefined}
                  />,
                  "className": "sphere-wrapper",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "planet-sphere",
                    "diameter": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "planet-name",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Col
                      md={8}
                      xs={24}
                    >
                      <Card
                        className="info-card"
                        title="Information"
                      >
                        <PlanetBasicInfo
                          planet={
                            Object {
                              "url": "planet_url",
                            }
                          }
                        />
                      </Card>
                    </Col>,
                    <Col
                      md={8}
                      xs={24}
                    >
                      <Card
                        className="info-card"
                        loading={true}
                        title="Residents"
                      />
                    </Col>,
                    <Col
                      md={8}
                      xs={24}
                    >
                      <Card
                        className="info-card"
                        loading={true}
                        title="Films"
                      />
                    </Col>,
                  ],
                  "gutter": 24,
                  "type": "flex",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Card
                        className="info-card"
                        title="Information"
                      >
                        <PlanetBasicInfo
                          planet={
                            Object {
                              "url": "planet_url",
                            }
                          }
                        />
                      </Card>,
                      "md": 8,
                      "xs": 24,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <PlanetBasicInfo
                          planet={
                            Object {
                              "url": "planet_url",
                            }
                          }
                        />,
                        "className": "info-card",
                        "title": "Information",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "planet": Object {
                            "url": "planet_url",
                          },
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Card
                        className="info-card"
                        loading={true}
                        title="Residents"
                      />,
                      "md": 8,
                      "xs": 24,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "info-card",
                        "loading": true,
                        "title": "Residents",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Card
                        className="info-card"
                        loading={true}
                        title="Films"
                      />,
                      "md": 8,
                      "xs": 24,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "info-card",
                        "loading": true,
                        "title": "Films",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
